### PR TITLE
search: create streaming progress UI

### LINKED ISSUE #15416: Streaming Search: create basic progress UI with fake data
Figma: https://www.figma.com/file/IyiXZIbPHK447NCXov0AvK/13928-Streaming-search?node-id=0%3A1

Progress API: https://github.com/sourcegraph/sourcegraph/issues/14977

### PR DESCRIPTION
**Pull request by limitedmage**
Adds progress indicator for streaming search (UI only, uses fake data)
- Progress indicator will be animated if streaming has not completed.
- A button will be shown if any skipped data is present.
- Clicking this button will show details of what has been skipped and the option to search again with suggested query modifications to expand the search.

Screen Recording 2020-11-09 at 4 30 38 PM

image

Fixes #15416

### CODE REVIEW

**Comment by keegancsmith** [comment]
I'll skip on reviewing since its a bunch of stuff I don't know. But the screenshots look great. Nice work!

**Review by attfarhan — APPROVED** [review]

File: client/web/src/search/results/streaming-progress/StreamingProgress.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            animation: gradient 3s linear infinite;
+        }
+    }
+
+    &__skipped {
+        border: none;
+        &.btn.btn-secondary {
+            color: var(--link-color);
+            background-color: transparent;
+            box-shadow: none;
```
Comment (felixfbecker) [inline client/web/src/search/results/streaming-progress/StreamingProgress.scss:31]: **issue**: I had raised this to @rrhyne in Figma too, but this is really bad for accessibility. With this line there is no way to focus this element visibly, breaking keyboard accessibility. If I tab through the buttons in that row, there is no indication of when this element is focused. We should never add such an override to remove focus outlines.

File: client/web/src/search/results/streaming-progress/StreamingProgress.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        }
```
[the remark this answers is not in the document]
Comment (limitedmage) [reply client/web/src/search/results/streaming-progress/StreamingProgress.scss:32]: This is exactly what I needed! Thank you!

File: client/web/src/search/results/streaming-progress/StreamingProgress.scss (hunk 1 above)
Comment (limitedmage) [reply client/web/src/search/results/streaming-progress/StreamingProgress.scss:31]: Thanks! I agree, switched to `btn-link` and left more default styles there. Changed it here: https://github.com/sourcegraph/sourcegraph/pull/16563

File: client/web/src/search/results/streaming-progress/StreamingProgressSkippedPopover.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                        ) : (
+                            <InformationOutlineIcon className="icon-inline mr-2" />
+                        )}
+                        <span>{skipped.title}</span>
+                    </h4>
+                    {skipped.message && <div className="mt-2">{skipped.message}</div>}
+                </Alert>
+            ))}
+            {progress.skipped.some(skipped => skipped.suggested) && (
+                <Form onSubmit={submitHandler}>
```
[the remark this answers is not in the document]
Comment (limitedmage) [reply client/web/src/search/results/streaming-progress/StreamingProgressSkippedPopover.tsx:51]: These are wrappers over Bootstrap's `Form`/`FormGroup`, which I'm using here for styling. I think we warrant some deeper discussion on when to use Reactstrap and when to use plain HTML with React classes. This is a very simple form with just basic validation, so it works fine either way.

**Pull request merged**